### PR TITLE
feat: lld - adds swap feature flag values to analytic events

### PR DESCRIPTION
**Pull request by cjordan-ledger**
### 📝 Description

Adds `isDexEnabled` to all analytics events in the swap flow. This is sourced from a feature flag, so our shared event properties object is now loaded via a hook.

### ❓ Context

- **Impacted projects**:  LLD
- **Linked resource(s)**:  https://ledgerhq.atlassian.net/browse/LIVE-5302

### ✅ Checklist

- [ ] **Test coverage** 
- [ ] **Atomic delivery** 
- [ ] **No breaking changes** 

### 🚀 Expectations to reach

_Please make sure you follow these [**Important Steps**](https://github.com/LedgerHQ/ledger-live/blob/develop/CONTRIBUTING.md#important-steps)._

_Pull Requests must pass the CI and be internally validated in order to be merged._

### CODE REVIEW

File: apps/ledger-live-desktop/src/renderer/screens/market/MarketCoinScreen/index.tsx [hunk 1]
```diff
@@ -224,7 +233,7 @@ export default function MarketCoinScreen() {
 
       startStakeFlow();
     },
-    [currency?.ticker, startStakeFlow],
+    [currency?.ticker, startStakeFlow, swapDefaultTrack],
```
Comment (sarneijim) [inline apps/ledger-live-desktop/src/renderer/screens/market/MarketCoinScreen/index.tsx:236]: There is a small bug here, we need to use here: "stakeDefaultTrack", not "swapDefaultTrack"

File: apps/ledger-live-desktop/src/renderer/screens/exchange/Swap2/utils/index.js [hunk 2]
```diff
@@ -1,17 +1,25 @@
...
-  flow: "swap",
+export const useGetSwapTrackingProperties = () => {
+  const swapShowDexQuotes = useFeature("swapShowDexQuotes");
+  return useMemo(
+    () => ({
+      swapVersion: SWAP_VERSION,
+      flow: "swap",
+      isDexEnabled: swapShowDexQuotes.enabled ?? false,
+    }),
+    [swapShowDexQuotes.enabled],
```
Comment (sarneijim) [inline apps/ledger-live-desktop/src/renderer/screens/exchange/Swap2/utils/index.js:21]: swapShowDexQuotes could be undefined.
```useFeature("swapShowDexQuotes") || {}```
or
```
swapShowDexQuotes?.enabled```

**Review by sarneijim — COMMENTED** [review]
@cpellizzari-ledger , could you confirm if you want to use this method?
https://ledger.slack.com/archives/C04DVPA0GG5/p1675250970948879

File: apps/ledger-live-desktop/src/renderer/screens/exchange/Swap2/utils/index.js (hunk 2 above)
Comment (cjordan-ledger) [reply apps/ledger-live-desktop/src/renderer/screens/exchange/Swap2/utils/index.js:21]: @sarneijim I think you have spotted the reason why tests are failing on CI and I've spent hours trying to figure out 👍

File: apps/ledger-live-desktop/src/renderer/screens/market/MarketCoinScreen/index.tsx (hunk 1 above)
Comment (cjordan-ledger) [reply apps/ledger-live-desktop/src/renderer/screens/market/MarketCoinScreen/index.tsx:236]: Great spot.

**Review by sarneijim — APPROVED** [review]

**Pull request merged**